testsuite/test_student_workflow.py

```diff
@@ -198,3 +200,28 @@ def test_anonymous(client, STC):
 
     response = client.get('/submission/')
     STC.assertRedirects(response, '/?next=/submission/')
+
+
+@pytest.mark.django_db
+def test_filesize_limit(client, base_db_setup, mock_evaluate):
+    (_, _, user, course, assignment) = base_db_setup
+
+    client.login(username=user.username, password='pw')
+
+    buff = BytesIO()
+    zip_archive = zipfile.ZipFile(buff, mode='w')
+    zip_archive.writestr('test.c', 'aaaa'*2**20)
+
+    upload = SimpleUploadedFile(
+        FILEPATH.name,
+        buff.read(),
+        content_type='application/zip',
+    )
+
+    client.post(
```

Comment:
Q: You said in the call that we can not see the error message right?
Thinking if we can see the error message on the webpage? (too big archive kind of stuff)

Reply:
The problem that I encountered is that I do not know, at the moment, how to talk from the upload handler to the rest of the code. Looks like I do not have a way to catch the exception `StopUpload`. I intend to look a bit into `handle_raw_input`, the docs say that i can return a tuple of `(POST, FILES)`, it might help.